integration_tests/cli/model-validation-cases.sh

```diff
@@ -0,0 +1,448 @@
+check_response_and_report "$result" "\"success\": true"
+
+test_divider
+
+# Query the updated model version 
+echo "Query the updated model version VID: $vid_1 PID: $pid_1 SV: $sv_1"
+result=$(dclcli query model get-model-version --vid=$vid_1 --pid=$pid_1 --softwareVersion=$sv_1)
+
+check_response_and_report "$result" "\"vid\": $vid_1"
+check_response_and_report "$result" "\"pid\": $pid_1"
+check_response_and_report "$result" "\"softwareVersion\": $sv_1"
+check_response_and_report "$result" "\"softwareVersionString\": \"1\""
+check_response_and_report "$result" "\"CDVersionNumber\": 1"
+check_response_and_report "$result" "\"softwareVersionValid\": false"
+check_response_and_report "$result" "\"minApplicableSoftwareVersion\": 1"
+check_response_and_report "$result" "\"maxApplicableSoftwareVersion\": 10"
+
+test_divider
+
+# Update the model version with few mutable fields and make sure all other fields are still the same
+echo "Update Device Model Version with few mutable fields and make sure all other fields are still the same for VID: $vid_1 PID: $pid_1 SV: $sv_1"
+result=$(echo "test1234" | dclcli tx model update-model-version --vid=$vid_1 --pid=$pid_1 --softwareVersion=$sv_1 --softwareVersionValid=true \
+--releaseNotesURL="https://release.url.info" --minApplicableSoftwareVersion=2 --from=$vendor_account_1 --yes)
+check_response_and_report "$result" "\"success\": true"
+
+test_divider
+
+# Query the updated model version 
+echo "Query the updated model version VID: $vid_1 PID: $pid_1 SV: $sv_1"
+result=$(dclcli query model get-model-version --vid=$vid_1 --pid=$pid_1 --softwareVersion=$sv_1)
+
+check_response_and_report "$result" "\"vid\": $vid_1"
+check_response_and_report "$result" "\"pid\": $pid_1"
+check_response_and_report "$result" "\"softwareVersion\": $sv_1"
+check_response_and_report "$result" "\"softwareVersionString\": \"1\""
+check_response_and_report "$result" "\"CDVersionNumber\": 1"
+check_response_and_report "$result" "\"softwareVersionValid\": true"
+check_response_and_report "$result" "\"minApplicableSoftwareVersion\": 2"
+check_response_and_report "$result" "\"maxApplicableSoftwareVersion\": 10"
+check_response_and_report "$result" "\"releaseNotesURL\": \"https://release.url.info\""
+
+
+
+sv_1=$RANDOM
+echo "Create a Device Model Version with all fields for VID: $vid_1 PID: $pid_1 SV: $sv_1"
+result=$(echo 'test1234' | dclcli tx model add-model-version --vid=$vid_1 --pid=$pid_1 --softwareVersion=$sv_1 \
+--softwareVersionString="1.0" --cdVersionNumber=21334 \
+--firmwareDigests="123456789012345678901234567890123456789012345678901234567890123" \
+--softwareVersionValid=true --otaURL="https://ota.url.info" --otaFileSize=123456789 \
+--otaChecksum="123456789012345678901234567890123456789012345678901234567890123" --releaseNotesURL="https://release.notes.url.info" \
+--otaChecksumType=1 --maxApplicableSoftwareVersion=32 --minApplicableSoftwareVersion=5   --from=$vendor_account_1 --yes)
```

Comment:
Do we (and should we) have validation that `minApplicableSoftwareVersion` is less than `maxApplicableSoftwareVersion`?